File: cpp_solutions/32. Longest Valid Parentheses.cpp
class Solution {
public:
    int longestValidParentheses(string s) {
        
        stack<int> stack;
        int ans = 0;
        stack.push(-1);
        for(int x=0;x<s.length();x++){
            if (s.at(x) == '('){
                stack.push(x);
            }
            else{
                stack.pop();
                if (stack.size()!=0){
                    ans = max(ans, x - stack.top());
                }
                else{
                    stack.push(x);
                }
            
            }
        }
        return ans;
        
    }
};

File: python3_solution_set1/32. Longest Valid Parentheses.py
##ss
class Solution:
    def longestValidParentheses(self, s: str) -> int:
        
        stack = []
        ans = 0
        stack.append(-1)
        
        for x in range(len(s)):
            if s[x] == '(':
                stack.append(x)
                
            else:
                stack.pop()
                if not stack:
                    stack.append(x)
                else:
                    ans = max(ans , x - stack[-1])
                    
        return ans
        
